valence-rs/valence
# Entity Rework

## Description

Closes #269
Closes #199 

- Removes `McEntity` and replaces it with bundles of components, one for each entity type.
- Tracked data types are now separate components rather than stuffing everything into a `TrackedData` enum.
- Tracked data is now cached in binary form within each entity, eliminating some work when entities enter the view of clients.
- Complete redesign of entity code generator.
- More docs for some components.
- Field bits are moved out of the entity extractor and into the valence entity module.
- Moved hitbox code to separate module.
- Refactor instance update systems to improve parallelism.

### TODOs
- [x] Update examples.
- [x] Update `default_event_handler`.

## Test Plan

Steps:
1. Check out the entity module docs with `cargo d --open`.
2. Run examples.
